setuptools/dist.py

```diff
@@ -964,7 +966,28 @@ def run_command(self, command):
         # Postpone defaults until all explicit configuration is considered
         # (setup() args, config files, command line and plugins)
 
-        super().run_command(command)
+        with self._override_get_fullname():
+            super().run_command(command)
+
+    @contextlib.contextmanager
+    def _override_get_fullname(self):
+        def _get_fullname_canonicalized(self):
+            return "{}-{}".format(
+                canonicalize_name(self.get_name()),
```

Comment:
This is the wrong canonicalization, as the test case later confirms.  

The hyphens were supposed to be replaced by underscores.

Reply:
Indeed! Thanks for the catch, fixed in bf33f79fee5ba88dba5dde8beb57ba03d856dc31.